src/Components/AdvancedSearch/Internal/AdvancedSearchResultDetailsList/AdvancedSearchResultDetailsList.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useCallback } from 'react';
+import {
+    IAdvancedSearchResultDetailsListProps,
+    IAdvancedSearchResultDetailsListStyleProps,
+    IAdvancedSearchResultDetailsListStyles
+} from './AdvancedSearchResultDetailsList.types';
+import { getStyles } from './AdvancedSearchResultDetailsList.styles';
+import {
+    classNamesFunction,
+    useTheme,
+    styled,
+    DetailsList,
+    DetailsListLayoutMode,
+    IColumn,
+    IDetailsListProps,
+    IconButton,
+    SelectionMode
+} from '@fluentui/react';
+import { IADTTwin } from '../../../../Models/Constants';
+import { useTranslation } from 'react-i18next';
+const getClassNames = classNamesFunction<
+    IAdvancedSearchResultDetailsListStyleProps,
+    IAdvancedSearchResultDetailsListStyles
+>();
+
+const AdvancedSearchResultDetailsList: React.FC<IAdvancedSearchResultDetailsListProps> = ({
+    filteredTwinsResult,
+    additionalColumns,
+    callbackFunction,
+    styles,
+    adapter,
+    theme,
+    locale
+}) => {
+    const { t } = useTranslation();
+
+    const classNames = getClassNames(styles, {
+        theme: useTheme()
+    });
+    const columns: IColumn[] = [
+        {
+            key: `twin-id + ${Math.random()}`,
+            name: t('twinId'),
+            fieldName: '$dtId',
+            minWidth: 100,
+            maxWidth: 200,
+            isResizable: true
+        },
+        {
+            key: 'properties',
+            name: t('properties'),
+            minWidth: 100,
+            maxWidth: 200,
+            isResizable: true
+        }
+    ];
+    const renderItemColumn: IDetailsListProps['onRenderItemColumn'] = (
+        item: any,
```

Comment:
we know the type of the item will be the same as the type you pass in on the collection of items so let's use that here too.